docs/glossary.md

```diff
@@ -0,0 +1,161 @@
+# Glossary
+
+## Anonymous branch
+
+An anonymous branch is a branch that doesn't have any [branches](#branch)
+pointing to it or to any of its descendants. Unlike Git, Jujutsu keeps such
+commits around until they are explicitly abandoned.
+
+## Backend
+
+A backend is an implementation of the storage layer. When unqualified, we
+typically mean "commit backend", which is the storage layer for commits. There
+are also pluggable backends for e.g. storing [the operation log](#operation-log)
+.
+
+## Branch
+
+A branch is a named pointer to a [commit](#commit). They automatically follow
+the commit if it gets [rewritten](#rewrite).
+
+See [here](branches.md) for details.
+
+## Change
+
+A change is a commit as it [evolves over time](#rewrite).
+
+## Change ID
+
+A change ID is a unique identifier for a [change](#change). They are typically
+16 bytes long. They are presented as hexadecimal numbers at the start of the
```

Comment:
I'd add: `... 16 bytes long and are often randomly generated`

And I'd replace the next sentences:

> By default, `jj log` presents them as a sequence of 12 letters in the `k-z` range, at the beginning of a line. These are actually hexadecimal numbers that use "digits" `k-z` instead of `0-9a-f`.

Reply:
Done

Reply:
I think you missed the last paragraph of my suggestion, where I reworded your description of hexadecimals. I still like the reworded version better, but feel free to go with either one.

Reply:
Done